### ISSUE TITLE
missing icons due to wrong svg content type

### ISSUE THREAD

**Issue by chavaroon** [microsoft/vscode#97564]
Issue Type: <b>Bug</b>

Try to set File Icon Theme to other than Seti (Visual Studio Code), and file icons will show no icons (or an empty tab space as icon). Also, somehow the root cause affected my 'Bookmarks' Extension which shows no icon. 

![Anotación 2020-05-11 200509](https://user-images.githubusercontent.com/65148411/81627025-d71a8780-93c2-11ea-91f4-0c02d3908f01.png)

I've downgraded VSCode to version 1.42, and the issue is not present there.
I'm pretty sure the issue is not related to VSCode version, as I have installed the latest version to another PC, and the issue is not present there.

I've also reinstalled VSCode as many as four or five times, deleting some tmp files from VSCode, but have not got rid of the issue.

VS Code version: Code 1.45.0 (d69a79b73808559a91206d73d7717ff5f798f23c, 2020-05-07T16:18:48.860Z)
OS version: Windows_NT x64 10.0.17763

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-8350U CPU @ 1.70GHz (8 x 1896)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>protected_video_decode: enabled<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>viz_display_compositor: enabled_on<br>viz_hit_test_surface_layer: disabled_off_ok<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|7.85GB (1.98GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (6)</summary>

Extension|Author (truncated)|Version
---|---|---
Bookmarks|ale|11.2.0
bracket-pair-colorizer|Coe|1.0.61
python|ms-|2020.4.76186
cpptools|ms-|0.28.0-insiders3
vscode-icons|vsc|10.1.1
srecord|xia|0.0.3

</details>

**Comment by aeschli** [comment]
Maybe a dup of https://github.com/microsoft/vscode/issues/92323

From https://github.com/microsoft/vscode/issues/92323#issuecomment-611045838:

Can you check 
- open regedit.exe
- search for `svg`
You should see Content Type image/svg+xml

**Comment by chavaroon** [comment]
Ok, saw #92323.

- Looked for `HKEY_CLASSES_ROOT\.svg`
- Found `Content Type` `text/xml`.
- Replaced `Content Type` with `image/svg+xml`, as suggested.
- **The issue is gone**.

Thank you, so much.!

**Comment by aeschli** [comment]
@deepak1556 @sandy081 This issue has been reported several times.

Any chance we can force Electron to use 'our' content types? For VSCode as a application, there's no point of using a custom content types for svg images.

**Comment by deepak1556** [comment]
Yup we can, we already do something close enough to block svg images from remote urls https://github.com/microsoft/vscode/blob/master/src/vs/code/electron-main/window.ts#L404-L413 . We can modify the condition a bit for local urls and edit the content-type in `onHeadersReceived`. The headers received in this method are from network layer before they have reached the renderer, so any response modified here will only be used for further processing by the renderer.

```
if (details.url.indexOf('.svg') > 0) {
   const uri = URI.parse(details.url);
   // I assume the svg are currently served over file
  // If its vscode-resource, changing according to the scheme
   if (uri && uri.scheme.match(/file/i)) {
      responseHeaders['content-type'] = 'image/svg+xml'
      return callback({ cancel: false, ...responseHeaders });
   }
}
```

**Comment by wmertens** [comment]
Triage: if this is indeed a dup of #92323, perhaps this issue can be closed?

**Comment by deepak1556** [comment]
@wmertens this issue tracks the solution to prevent svg being loaded with incorrect Content-Type headers.

**Comment by aeschli** [comment]
I still see the issue in 
Version: 1.53.0-insider (user setup) Commit: 5a52bc29d5e9bc419077552d336ea26d904299fa Date: 2021-01-29T05:13:48.533Z
Electron: 11.2.1
OS: Windows_NT x64 10.0.19042

- open `regedit`, search for `.svg`, change `image/svg+xml` to `text/xml`
- open code-insiders, switch to the minimal file icon theme -> file icons are empty

**Comment by jblievremont** [comment]
I have an issue that looks similar with VSCode 1.53.2 on Ubuntu 20.04: the SVG icons I use in an extension, on both tree view items and inside webviews, don't show.

Their PNG equivalents show, so it seems that SVG is specifically blocked (confirmed by the Webview Development Tools that show a download size of "0 B" for the SVG).

Could it be related, or shall I open another ticket?

**Comment by OmarOdaini** [comment]
Hi guys, can someone point me to a similar fix on mac. I have this issue with my docker extension. thanks

**Comment by rzhao271** [comment]
The current mitigation is https://github.com/microsoft/vscode/issues/97564#issuecomment-627351149
I used those steps to intentionally change the type back to `text/xml`, and I noticed that the issue still repros.
